### PR TITLE
Implement iterable for headers

### LINKED ISSUE #11897: Implement the Headers API
Specification: https://fetch.spec.whatwg.org/#headers-class
Documentation: http://doc.servo.org/servo/script/dom/index.html#adding-a-new-dom-interface

This is assigned to our RGSoC team, so please talk with me before starting to work on any part of it.

### PR DESCRIPTION
**Pull request by jeenalee**
These patches make Headers iterable based on #12819. 

The expected wpt results are updated as well. 

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #11897 (github issue number if applicable). 

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because web platform tests for Headers already exist. Expected test results are updated with this PR.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12998)

### CODE REVIEW

File: components/script/dom/headers.rs [hunk 1]
```diff
@@ -217,6 +225,41 @@ impl Headers {
...
+
+    fn get_value_at_index(&self, n: u32) -> ByteString {
+        let sorted_header_vec = self.sort_header_list();
+        let value = sorted_header_vec[n as usize].1.clone();
+        ByteString::new(value.into_bytes().to_vec())
+    }
+
+    fn get_key_at_index(&self, n: u32) -> ByteString {
+        let sorted_header_vec = self.sort_header_list();
+        let key = sorted_header_vec[n as usize].1.clone();
```
Comment (jdm) [inline components/script/dom/headers.rs]: I suspect this should be `.0` instead... do any test results change when that's corrected?

Comment (jeenalee) [inline components/script/dom/headers.rs]: Oh strange! Yes you are correct. I had it as 0, but I don't know what happened there... I'll fix that right now.

**Comment by jdm** [comment]
There are a bunch of unrelated changes in the commit that implements Iterable; it would be best to remove those (at least into separate commits that can be removed from this PR before merging).

**Comment by jdm** [comment]
#12819 should be merging within the next couple hours, so feel free to put this all until you can rebase your changes against master.

**Comment by jeenalee** [comment]
Ok. I must have made mistakes with git. It wasn't my intention to group all the changes in the headers with the iterable implementation commit. I'll wait until the #12819 is merged into master.

**Comment by jeenalee** [comment]
@jdm Rebased!

**Comment by jdm** [comment]
@bors-servo: r+
Nice work!

**Pull request merged**